feature/experimental/p4rt/ate_tests/traceroute_packetin_test/packetin_test.go

```diff
@@ -148,79 +147,75 @@ func testPacketIn(ctx context.Context, t *testing.T, args *testArgs, IsIpv4 bool
 
 	// Send Traceroute traffic from ATE
 	srcEndPoint := args.top.Interfaces()[atePort1.Name]
-
+	pktOut := testTraffic(t, args.ate, args.packetIO.GetTrafficFlow(args.ate, IsIpv4, 1, 300, 2), srcEndPoint, 10)
 	packetInTests := []struct {
-		desc       string
-		client     *p4rt_client.P4RTClient
-		expectPass bool
+		desc     string
+		client   *p4rt_client.P4RTClient
+		wantPkts int
 	}{{
-		desc:       "PacketIn to Primary Controller",
-		client:     leader,
-		expectPass: true,
+		desc:     "PacketIn to Primary Controller",
+		client:   leader,
+		wantPkts: pktOut,
 	}, {
-		desc:       "PacketIn to Secondary Controller",
-		client:     follower,
-		expectPass: false,
+		desc:     "PacketIn to Secondary Controller",
+		client:   follower,
+		wantPkts: 0,
 	}}
 
 	t.Log("TTL/HopLimit 1")
-	testTraffic(t, args.ate, args.packetIO.GetTrafficFlow(args.ate, IsIpv4, 1, 300, 2), srcEndPoint, 10)
 	for _, test := range packetInTests {
 		t.Run(test.desc, func(t *testing.T) {
 			// Extract packets from PacketIn message sent to p4rt client
-			packets := fetchPackets(ctx, t, test.client, 20)
+			packets, err := fetchPackets(ctx, t, test.client, test.wantPkts)
+			if err != nil {
+				t.Errorf("Unexpected error on fetchPackets: %v", err)
+			}
 
-			if !test.expectPass {
-				if len(packets) > 0 {
-					t.Fatalf("Unexpected packets received.")
-					return
-				}
+			if got, want := len(packets), test.wantPkts; got != want {
+				t.Errorf("Number of PacketIn, got: %d, want: %d", got, want)
+			}
+			if test.wantPkts == 0 {
 				return
-			} else {
-				if len(packets) == 0 {
-					t.Fatalf("There are no packets received.")
-					return
-				}
-				t.Logf("Start to decode packet and compare with expected packets.")
-				wantPacket := args.packetIO.GetPacketTemplate()
-				for _, packet := range packets {
-					if packet != nil {
-						if wantPacket.TTL != nil {
-							//TTL/HopLimit comparison for IPV4 & IPV6
-							if IsIpv4 {
-								captureTTL := decodePacket4(t, packet.Pkt.GetPayload())
-								if captureTTL != TTL1 {
-									t.Fatalf("Packet in PacketIn message is not matching wanted packet=IPV4 TTL1")
-								}
-
-							} else {
-								captureHopLimit := decodePacket6(t, packet.Pkt.GetPayload())
-								if captureHopLimit != HopLimit1 {
-									t.Fatalf("Packet in PacketIn message is not matching wanted packet=IPV6 HopLimit1")
-								}
+			}
+			t.Logf("Start to decode packet and compare with expected packets.")
+			wantPacket := args.packetIO.GetPacketTemplate()
+			for _, packet := range packets {
+				if packet != nil {
+					if wantPacket.TTL != nil {
+						//TTL/HopLimit comparison for IPV4 & IPV6
+						if IsIpv4 {
```

Comment:
IsIpv4 -> isIPv4?

Go initialism. I'm also not sure why the symbol is exported?